src/main/java/com/metamx/common/StreamUtils.java

```diff
@@ -99,7 +103,9 @@ public static long copyToFileAndClose(InputStream is, File file, long timeout) t
   public static long copyAndClose(InputStream is, OutputStream os) throws IOException
   {
     try {
-      return ByteStreams.copy(is, os);
+      final long retval = ByteStreams.copy(is, os);
+      os.flush();
```

Comment:
Won't work for filtered output stream due to http://hg.openjdk.java.net/jdk8/jdk8/jdk/rev/759aa847dcaf